```shell
python "/share/compling/projects/sanpi/script/sample_pickle.py" \
    -p "${DATA_DIR}neg-bigrams_thr0-001p.35f.pkl.gz" \
    -f 'neg_deprel!=neg' -f 'neg_deprel!=advmod' -f 'neg_deprel!=nsubj' -f 'neg_deprel!=case' \
    -f 'neg_deprel!=cc' -f 'neg_deprel!=cc:preconj' -f 'neg_deprel!=cop' -f 'neg_deprel!=det' \
    -c 'neg_deprel' -c 'neg_form_lower' -c 'adv_form_lower' -c 'adj_form_lower' -c 'text_window' \
    -f 'neg_head==ADJ' -N 10000 -q -C > "${DATA_DIR}sample-35f_unk-deprel_adj-head.10k.csv"
```

In [2]:
import pandas as pd
from pathlib import Path
from utils import sample_pickle

DATA_DIR=Path("/share/compling/data/sanpi/4_post-processed/RBdirect")
COLUMNS=['neg_deprel', 'neg_form_lower', 'adv_form_lower', 'adj_form_lower', 'text_window']
HEAD = 'ADJ'
N_FILES_PROC = 35
N_ROWS = 10000

#> Loaded variable 'df' from URI: /share/compling/data/sanpi/4_post-processed/RBdirect/sample-35f_unk-deprel_adj-head.1k.csv
# df = pd.read_csv(r'/share/compling/data/sanpi/4_post-processed/RBdirect/sample-35f_unk-deprel_adj-head.5k.csv')


## Define permitted `neg_deprel` values by head node

From [Oct 23 log](https://gist.github.com/AndreaTheLinguist/ef895a6e8602a9a82ff729f219bc5628#running-notes-monday-october-23-2023):

### `ADJ` _head_

i.e., `NEG` ⬅️ `ADJ`

> | permit ✅                                                         | prohibit ⛔                                              |
> |--------------------------------------------------------------------|----------------------------------------------------------|
> | `neg`, `advmod`, `nsubj`, `case`, `cop`, `det`, `cc`, `cc:preconj` | `advcl`, `amod`, `discourse`, `prep`, `obl`, `obl:npmod` |

In [3]:
OK_DEPREL = {
    'ADJ': ['neg', 'advmod', 'nsubj', 'case', 'cop', 'det', 'cc', 'cc:preconj'], 
    'NEG': ['amod', 'ccomp', 'csubj', 'dep', 'infomod', 'pcomp', 'pobj']
    }

## Load data from composite cleaned and filtered table of hit info

### For 10 corpus parts and only 8 rows first, to illustrate usage of `sample_pickle`

Note that any sorting (as specified by `sort_by` parameter) is applie _after_ sample is drawn

In [4]:
def set_input_path(n_files: int = N_FILES_PROC) -> Path:
    return DATA_DIR.joinpath(f'neg-bigrams_thr0-001p.{n_files}f.pkl.gz')


def set_filters(head: str = HEAD,
                ok_deprel: dict = OK_DEPREL) -> list:

    # filters = [f'neg_deprel!=neg', 'neg_deprel!=advmod', 'neg_deprel!=nsubj', 'neg_deprel!=case',
    #            'neg_deprel!=cop', 'neg_deprel!=det', 'neg_deprel!=cc', 'neg_deprel!=cc:preconj'
    #            ]
    filters = [f'neg_deprel!={rel}' for rel in ok_deprel[head]]
    return filters + [f'neg_head=={head}']


def get_sample(data_path, filters, print_table, sample_size) -> pd.DataFrame:
    return sample_pickle(
        data_path=data_path,
        sample_size=sample_size,
        columns=COLUMNS,
        filters=filters,
        markdown=True,
        sort_by='neg_deprel',
        print_sample=print_table
    )


df = get_sample(data_path=set_input_path(n_files=10),
                filters=set_filters(),
                print_table=True,
                sample_size=8)


## Sampling from `/share/compling/data/sanpi/4_post-processed/RBdirect/neg-bigrams_thr0-001p.10f.pkl.gz`

- *filtering rows...*
  - ✓ Applied filter: `neg_deprel!=neg`
  - ✓ Applied filter: `neg_deprel!=advmod`
  - ✓ Applied filter: `neg_deprel!=nsubj`
  - ✓ Applied filter: `neg_deprel!=case`
  - ✓ Applied filter: `neg_deprel!=cop`
  - ✓ Applied filter: `neg_deprel!=det`
  - ✓ Applied filter: `neg_deprel!=cc`
  - ✓ Applied filter: `neg_deprel!=cc:preconj`
  - ✓ Applied filter: `neg_head==ADJ`

- *selecting columns...*

### 8 random rows matching filter(s) from `neg-bigrams_thr0-001p.10f.pkl.gz`

| hit_id                                   | neg_deprel   | neg_form_lower   | adv_form_lower   | adj_form_lower   | text_window                                                                                                        |
|:-----------------------------------------|:-------------|:-----------------|:-----------------|:-----------------|:---------------------------------------------


## Sampling from `/share/compling/data/sanpi/4_post-processed/RBdirect/neg-bigrams_thr0-001p.10f.pkl.gz`

- *filtering rows...*
  - ✓ Applied filter: `neg_deprel!=neg`
  - ✓ Applied filter: `neg_deprel!=advmod`
  - ✓ Applied filter: `neg_deprel!=nsubj`
  - ✓ Applied filter: `neg_deprel!=case`
  - ✓ Applied filter: `neg_deprel!=cop`
  - ✓ Applied filter: `neg_deprel!=det`
  - ✓ Applied filter: `neg_deprel!=cc`
  - ✓ Applied filter: `neg_deprel!=cc:preconj`
  - ✓ Applied filter: `neg_head==ADJ`

- *selecting columns...*

### 8 random rows matching filter(s) from `neg-bigrams_thr0-001p.10f.pkl.gz`

| hit_id                                   | neg_deprel   | neg_form_lower   | adv_form_lower   | adj_form_lower   | text_window                                                                                                        |
|:-----------------------------------------|:-------------|:-----------------|:-----------------|:-----------------|:-------------------------------------------------------------------------------------------------------------------|
| apw_eng_20020208_0776_6:24-36-37         | dep          | without          | more             | difficult        | this course because , without them , the new equilibrium that we are seeking will be more difficult to attain , '' |
| nyt_eng_19990416_0358_28:08-11-12        | dep          | neither          | particularly     | complex          | n't easy , but neither was it particularly complex .                                                               |
| apw_eng_20090522_0774_13:2-3-4           | dep          | no               | more             | clear            | with no more clear chances for sprinters ,                                                                         |
| apw_eng_19970501_1204_15:24-25-26        | dep          | no               | longer           | afraid           | the Conservatives _ and no longer afraid of Labor .                                                                |
| pcc_eng_20_099.6848_x1594566_07:29-31-32 | mark         | without          | as               | blatant          | of the scenes ( without being as blatant as traditional porn )                                                     |
| pcc_eng_06_012.5761_x0187070_041:6-8-9   | mark         | without          | too              | sweet            | is rich and complex without being too sweet and comes in a                                                         |
| pcc_eng_19_050.8210_x0804180_82:24-26-27 | mark         | without          | consciously      | aware            | greater quantity , sometimes without being consciously aware that learning is taking                               |
| pcc_eng_18_046.7718_x0740811_07:25-27-28 | mark         | without          | too              | overwhelming     | the students to shore without being too overwhelming .                                                             |


In [5]:
def clean_data(df):
    if 'hit_id' in df.columns:
        hit_id = df.hit_id
        df = df.set_index('hit_id')
    else: 
        hit_id = df.index.to_series()

    df['corpus'] = hit_id.str.split('_eng_').str.get(0)
    df['match_ix'] = hit_id.str.rsplit(':',1).str.get(1)
    return df

df_clean = clean_data(df.copy())
df_clean.head()

neg_deprel neg_form_lower  \
hit_id                                                               
apw_eng_20020208_0776_6:24-36-37                dep        without   
nyt_eng_19990416_0358_28:08-11-12               dep        neither   
apw_eng_20090522_0774_13:2-3-4                  dep             no   
apw_eng_19970501_1204_15:24-25-26               dep             no   
pcc_eng_20_099.6848_x1594566_07:29-31-32       mark        without   

                                         adv_form_lower adj_form_lower  \
hit_id                                                                   
apw_eng_20020208_0776_6:24-36-37                   more      difficult   
nyt_eng_19990416_0358_28:08-11-12          particularly        complex   
apw_eng_20090522_0774_13:2-3-4                     more          clear   
apw_eng_19970501_1204_15:24-25-26                longer         afraid   
pcc_eng_20_099.6848_x1594566_07:29-31-32             as        blatant   

                                                                                text_window  \
hit_id                                                                                        
apw_eng_20020208_0776_6:24-36-37          this course because , without them , the new e...   
nyt_eng_19990416_0358_28:08-11-12         n't easy , but neither was it particularly com...   
apw_eng_20090522_0774_13:2-3-4                   with no more clear chances for sprinters ,   
apw_eng_19970501_1204_15:24-25-26         the Conservatives _ and no longer afraid of La...   
pcc_eng_20_099.6848_x1594566_07:29-31-32  of the scenes ( without being as blatant as tr...   

                                         corpus  match_ix  
hit_id                                                     
apw_eng_20020208_0776_6:24-36-37            apw  24-36-37  
nyt_eng_19990416_0358_28:08-11-12           nyt  08-11-12  
apw_eng_20090522_0774_13:2-3-4              apw     2-3-4  
apw_eng_19970501_1204_15:24-25-26           apw  24-25-26  
pcc_eng_20_099.6848_x1594566_07:29-31-32    pcc  29-31-32

### Now with larger sample and printing suppressed

In [19]:
df = get_sample(data_path=set_input_path(n_files=N_FILES_PROC),
                filters=set_filters(),
                print_table=False,
                sample_size=10000)
df_clean = clean_data(df.copy())


## Sampling from `/share/compling/data/sanpi/4_post-processed/RBdirect/neg-bigrams_thr0-001p.35f.pkl.gz`

- *filtering rows...*
  - ✓ Applied filter: `neg_deprel!=neg`
  - ✓ Applied filter: `neg_deprel!=advmod`
  - ✓ Applied filter: `neg_deprel!=nsubj`
  - ✓ Applied filter: `neg_deprel!=case`
  - ✓ Applied filter: `neg_deprel!=cop`
  - ✓ Applied filter: `neg_deprel!=det`
  - ✓ Applied filter: `neg_deprel!=cc`
  - ✓ Applied filter: `neg_deprel!=cc:preconj`
  - ✓ Applied filter: `neg_head==ADJ`

- *selecting columns...*

### 10000 random rows matching filter(s) from `neg-bigrams_thr0-001p.35f.pkl.gz`




## Sampling from `/share/compling/data/sanpi/4_post-processed/RBdirect/neg-bigrams_thr0-001p.35f.pkl.gz`

- *filtering rows...*
  - ✓ Applied filter: `neg_deprel!=neg`
  - ✓ Applied filter: `neg_deprel!=advmod`
  - ✓ Applied filter: `neg_deprel!=nsubj`
  - ✓ Applied filter: `neg_deprel!=case`
  - ✓ Applied filter: `neg_deprel!=cop`
  - ✓ Applied filter: `neg_deprel!=det`
  - ✓ Applied filter: `neg_deprel!=cc`
  - ✓ Applied filter: `neg_deprel!=cc:preconj`
  - ✓ Applied filter: `neg_head==ADJ`

- *selecting columns...*

### 10000 random rows matching filter(s) from `neg-bigrams_thr0-001p.35f.pkl.gz`

In [20]:
df_clean

neg_deprel neg_form_lower  \
hit_id                                                             
nyt_eng_19981129_0157_24:48-51-52            attr        nothing   
nyt_eng_19960302_0173_15:09-22-23            attr        nothing   
nyt_eng_20000806_0185_4:04-10-11             attr        nothing   
nyt_eng_19990909_0107_2:7-8-9                attr           none   
pcc_eng_22_044.1204_x0696858_08:4-5-6         aux          ain't   
...                                           ...            ...   
pcc_eng_10_053.8207_x0854279_115:2-8-9   vocative         nobody   
pcc_eng_19_027.9735_x0435427_58:1-5-6    vocative        nothing   
pcc_eng_11_057.3830_x0912194_107:2-6-7   vocative        nothing   
pcc_eng_07_061.3782_x0975972_094:1-5-6   vocative        nothing   
pcc_eng_06_034.7520_x0545912_06:1-5-6    vocative           none   

                                          adv_form_lower adj_form_lower  \
hit_id                                                                    
nyt_eng_19981129_0157_24:48-51-52              certainly       relative   
nyt_eng_19960302_0173_15:09-22-23       constitutionally          unfit   
nyt_eng_20000806_0185_4:04-10-11                somewhat           akin   
nyt_eng_19990909_0107_2:7-8-9                        too        pleased   
pcc_eng_22_044.1204_x0696858_08:4-5-6                 so          elite   
...                                                  ...            ...   
pcc_eng_10_053.8207_x0854279_115:2-8-9              more        shocked   
pcc_eng_19_027.9735_x0435427_58:1-5-6              quite       thankful   
pcc_eng_11_057.3830_x0912194_107:2-6-7        completely           fine   
pcc_eng_07_061.3782_x0975972_094:1-5-6          probably         asleep   
pcc_eng_06_034.7520_x0545912_06:1-5-6                n't        serious   

                                                                              text_window  \
hit_id                                                                                      
nyt_eng_19981129_0157_24:48-51-52       would prove to be nothing special , certainly ...   
nyt_eng_19960302_0173_15:09-22-23       supporters admit she is nothing if not indepen...   
nyt_eng_20000806_0185_4:04-10-11        the return was nothing but a beat down _ somew...   
nyt_eng_19990909_0107_2:7-8-9           the Catholic Church being none too pleased , o...   
pcc_eng_22_044.1204_x0696858_08:4-5-6          I guess you ain't so elite after all , are   
...                                                                                   ...   
pcc_eng_10_053.8207_x0854279_115:2-8-9  And nobody , nobody could have been more shock...   
pcc_eng_19_027.9735_x0435427_58:1-5-6        Nothing , I am quite thankful for the way my   
pcc_eng_11_057.3830_x0912194_107:2-6-7            " Nothing , I 'm completely fine Emma .   
pcc_eng_07_061.3782_x0975972_094:1-5-6    Nothing , she 's probably asleep , and when she   
pcc_eng_06_034.7520_x0545912_06:1-5-6                         None , it was n't serious .   

                                       corpus  match_ix  
hit_id                                                   
nyt_eng_19981129_0157_24:48-51-52         nyt  48-51-52  
nyt_eng_19960302_0173_15:09-22-23         nyt  09-22-23  
nyt_eng_20000806_0185_4:04-10-11          nyt  04-10-11  
nyt_eng_19990909_0107_2:7-8-9             nyt     7-8-9  
pcc_eng_22_044.1204_x0696858_08:4-5-6     pcc     4-5-6  
...                                       ...       ...  
pcc_eng_10_053.8207_x0854279_115:2-8-9    pcc     2-8-9  
pcc_eng_19_027.9735_x0435427_58:1-5-6     pcc     1-5-6  
pcc_eng_11_057.3830_x0912194_107:2-6-7    pcc     2-6-7  
pcc_eng_07_061.3782_x0975972_094:1-5-6    pcc     1-5-6  
pcc_eng_06_034.7520_x0545912_06:1-5-6     pcc     1-5-6  

[10000 rows x 7 columns]

In [26]:
deprel_counts=df_clean.value_counts('neg_deprel', normalize=True)
print(deprel_counts.multiply(100).round(2).to_frame('% of 10k ADJ-head sample').to_markdown(floatfmt=',.2f'))

| neg_deprel   |   % of 10k ADJ-head sample |
|:-------------|---------------------------:|
| mark         |                      52.96 |
| dep          |                      40.49 |
| conj         |                       3.25 |
| preconj      |                       1.61 |
| parataxis    |                       0.52 |
| aux          |                       0.41 |
| dislocated   |                       0.16 |
| rel          |                       0.13 |
| ccomp        |                       0.10 |
| vocative     |                       0.09 |
| nsubjpass    |                       0.07 |
| reparandum   |                       0.06 |
| attr         |                       0.04 |
| obj          |                       0.03 |
| punct        |                       0.03 |
| expl         |                       0.02 |
| dobj         |                       0.02 |
| compound     |                       0.01 |


| neg_deprel   |   % of 10k ADJ-head sample |
|:-------------|---------------------------:|
| mark         |                      52.96 |
| dep          |                      40.49 |
| conj         |                       3.25 |
| preconj      |                       1.61 |
| parataxis    |                       0.52 |
| aux          |                       0.41 |
| dislocated   |                       0.16 |
| rel          |                       0.13 |
| ccomp        |                       0.10 |
| vocative     |                       0.09 |
| nsubjpass    |                       0.07 |
| reparandum   |                       0.06 |
| attr         |                       0.04 |
| obj          |                       0.03 |
| punct        |                       0.03 |
| expl         |                       0.02 |
| dobj         |                       0.02 |
| compound     |                       0.01 |

In [28]:
for g, d in df_clean[['neg_deprel', 'neg_form_lower', 'corpus']].groupby('neg_deprel'):
    vc = d.value_counts(normalize=True).multiply(100).round(2).to_frame(
        '% of `neg_deprel`')
    vc['count']=d.value_counts()
    vc = vc.reset_index().iloc[:, 1:].set_index('neg_form_lower')

    print(f'`neg_deprel == {g}`: _{len(d)} total in 10k sample ({round(len(d)/len(df_clean) * 100, 2)}%)_')
    print(vc.to_markdown(floatfmt=',.1f'))
    print('\n---\n')

`neg_deprel == attr`: _4 total in 10k sample (0.04%)_
| neg_form_lower   | corpus   |   % of `neg_deprel` |   count |
|:-----------------|:---------|--------------------:|--------:|
| nothing          | nyt      |                75.0 |       3 |
| none             | nyt      |                25.0 |       1 |

---

`neg_deprel == aux`: _41 total in 10k sample (0.41%)_
| neg_form_lower   | corpus   |   % of `neg_deprel` |   count |
|:-----------------|:---------|--------------------:|--------:|
| ain't            | pcc      |                90.2 |      37 |
| aint             | pcc      |                 9.8 |       4 |

---

`neg_deprel == ccomp`: _10 total in 10k sample (0.1%)_
| neg_form_lower   | corpus   |   % of `neg_deprel` |   count |
|:-----------------|:---------|--------------------:|--------:|
| not              | nyt      |                40.0 |       4 |
| not              | apw      |                20.0 |       2 |
| none             | nyt      |                10.0 |    

`neg_deprel == attr`: _4 total in 10k sample (0.04%)_
| neg_form_lower   | corpus   |   % of `neg_deprel` |   count |
|:-----------------|:---------|--------------------:|--------:|
| nothing          | nyt      |                75.0 |       3 |
| none             | nyt      |                25.0 |       1 |

---

`neg_deprel == aux`: _41 total in 10k sample (0.41%)_
| neg_form_lower   | corpus   |   % of `neg_deprel` |   count |
|:-----------------|:---------|--------------------:|--------:|
| ain't            | pcc      |                90.2 |      37 |
| aint             | pcc      |                 9.8 |       4 |

---

`neg_deprel == ccomp`: _10 total in 10k sample (0.1%)_
| neg_form_lower   | corpus   |   % of `neg_deprel` |   count |
|:-----------------|:---------|--------------------:|--------:|
| not              | nyt      |                40.0 |       4 |
| not              | apw      |                20.0 |       2 |
| none             | nyt      |                10.0 |       1 |
| not              | pcc      |                10.0 |       1 |
| nothing          | nyt      |                10.0 |       1 |
| nothing          | pcc      |                10.0 |       1 |

---

`neg_deprel == compound`: _1 total in 10k sample (0.01%)_
| neg_form_lower   | corpus   |   % of `neg_deprel` |   count |
|:-----------------|:---------|--------------------:|--------:|
| none             | pcc      |               100.0 |       1 |

---

`neg_deprel == conj`: _325 total in 10k sample (3.25%)_
| neg_form_lower   | corpus   |   % of `neg_deprel` |   count |
|:-----------------|:---------|--------------------:|--------:|
| not              | nyt      |                74.8 |     243 |
| not              | apw      |                22.5 |      73 |
| never            | nyt      |                 0.6 |       2 |
| not              | pcc      |                 0.6 |       2 |
| nothing          | nyt      |                 0.6 |       2 |
| few              | pcc      |                 0.3 |       1 |
| none             | nyt      |                 0.3 |       1 |
| nothing          | pcc      |                 0.3 |       1 |

---

`neg_deprel == dep`: _4049 total in 10k sample (40.49%)_
| neg_form_lower   | corpus   |   % of `neg_deprel` |   count |
|:-----------------|:---------|--------------------:|--------:|
| no               | nyt      |                40.1 |    1625 |
| not              | nyt      |                34.4 |    1392 |
| no               | apw      |                 7.5 |     303 |
| not              | apw      |                 6.3 |     255 |
| nothing          | nyt      |                 3.6 |     147 |
| none             | nyt      |                 2.0 |      82 |
| neither          | nyt      |                 1.9 |      77 |
| neither          | apw      |                 1.0 |      41 |
| nothing          | apw      |                 0.9 |      35 |
| few              | nyt      |                 0.5 |      19 |
| none             | apw      |                 0.4 |      15 |
| never            | nyt      |                 0.3 |      14 |
| without          | nyt      |                 0.3 |      13 |
| nor              | nyt      |                 0.2 |      10 |
| without          | apw      |                 0.1 |       6 |
| few              | apw      |                 0.1 |       3 |
| rarely           | nyt      |                 0.1 |       3 |
| n't              | nyt      |                 0.1 |       2 |
| hardly           | nyt      |                 0.1 |       2 |
| barely           | apw      |                 0.0 |       1 |
| nobody           | nyt      |                 0.0 |       1 |
| never            | apw      |                 0.0 |       1 |
| neither          | pcc      |                 0.0 |       1 |
| nobody           | apw      |                 0.0 |       1 |

---

`neg_deprel == dislocated`: _16 total in 10k sample (0.16%)_
| neg_form_lower   | corpus   |   % of `neg_deprel` |   count |
|:-----------------|:---------|--------------------:|--------:|
| nothing          | pcc      |                81.2 |      13 |
| none             | pcc      |                18.8 |       3 |

---

`neg_deprel == dobj`: _2 total in 10k sample (0.02%)_
| neg_form_lower   | corpus   |   % of `neg_deprel` |   count |
|:-----------------|:---------|--------------------:|--------:|
| neither          | nyt      |                50.0 |       1 |
| nothing          | apw      |                50.0 |       1 |

---

`neg_deprel == expl`: _2 total in 10k sample (0.02%)_
| neg_form_lower   | corpus   |   % of `neg_deprel` |   count |
|:-----------------|:---------|--------------------:|--------:|
| nit              | pcc      |               100.0 |       2 |

---

`neg_deprel == mark`: _5296 total in 10k sample (52.96%)_
| neg_form_lower   | corpus   |   % of `neg_deprel` |   count |
|:-----------------|:---------|--------------------:|--------:|
| without          | pcc      |                99.4 |    5265 |
| without          | nyt      |                 0.2 |      12 |
| no               | pcc      |                 0.1 |       6 |
| without          | apw      |                 0.1 |       5 |
| w/o              | pcc      |                 0.1 |       4 |
| na               | pcc      |                 0.1 |       3 |
| aint             | pcc      |                 0.0 |       1 |

---

`neg_deprel == nsubjpass`: _7 total in 10k sample (0.07%)_
| neg_form_lower   | corpus   |   % of `neg_deprel` |   count |
|:-----------------|:---------|--------------------:|--------:|
| none             | nyt      |                42.9 |       3 |
| nothing          | nyt      |                42.9 |       3 |
| few              | nyt      |                14.3 |       1 |

---

`neg_deprel == obj`: _3 total in 10k sample (0.03%)_
| neg_form_lower   | corpus   |   % of `neg_deprel` |   count |
|:-----------------|:---------|--------------------:|--------:|
| neither          | pcc      |                66.7 |       2 |
| no               | pcc      |                33.3 |       1 |

---

`neg_deprel == parataxis`: _52 total in 10k sample (0.52%)_
| neg_form_lower   | corpus   |   % of `neg_deprel` |   count |
|:-----------------|:---------|--------------------:|--------:|
| not              | nyt      |                75.0 |      39 |
| not              | apw      |                17.3 |       9 |
| aint             | pcc      |                 1.9 |       1 |
| few              | apw      |                 1.9 |       1 |
| few              | pcc      |                 1.9 |       1 |
| none             | pcc      |                 1.9 |       1 |

---

`neg_deprel == preconj`: _161 total in 10k sample (1.61%)_
| neg_form_lower   | corpus   |   % of `neg_deprel` |   count |
|:-----------------|:---------|--------------------:|--------:|
| neither          | nyt      |                89.4 |     144 |
| neither          | apw      |                10.6 |      17 |

---

`neg_deprel == punct`: _3 total in 10k sample (0.03%)_
| neg_form_lower   | corpus   |   % of `neg_deprel` |   count |
|:-----------------|:---------|--------------------:|--------:|
| ain't            | pcc      |               100.0 |       3 |

---

`neg_deprel == rel`: _13 total in 10k sample (0.13%)_
| neg_form_lower   | corpus   |   % of `neg_deprel` |   count |
|:-----------------|:---------|--------------------:|--------:|
| without          | nyt      |                69.2 |       9 |
| without          | apw      |                30.8 |       4 |

---

`neg_deprel == reparandum`: _6 total in 10k sample (0.06%)_
| neg_form_lower   | corpus   |   % of `neg_deprel` |   count |
|:-----------------|:---------|--------------------:|--------:|
| no               | pcc      |                83.3 |       5 |
| not              | pcc      |                16.7 |       1 |

---

`neg_deprel == vocative`: _9 total in 10k sample (0.09%)_
| neg_form_lower   | corpus   |   % of `neg_deprel` |   count |
|:-----------------|:---------|--------------------:|--------:|
| nothing          | pcc      |                44.4 |       4 |
| none             | pcc      |                33.3 |       3 |
| nobody           | pcc      |                22.2 |       2 |

In [31]:
df_clean

neg_deprel neg_form_lower  \
hit_id                                                             
nyt_eng_19981129_0157_24:48-51-52            attr        nothing   
nyt_eng_19960302_0173_15:09-22-23            attr        nothing   
nyt_eng_20000806_0185_4:04-10-11             attr        nothing   
nyt_eng_19990909_0107_2:7-8-9                attr           none   
pcc_eng_22_044.1204_x0696858_08:4-5-6         aux          ain't   
...                                           ...            ...   
pcc_eng_10_053.8207_x0854279_115:2-8-9   vocative         nobody   
pcc_eng_19_027.9735_x0435427_58:1-5-6    vocative        nothing   
pcc_eng_11_057.3830_x0912194_107:2-6-7   vocative        nothing   
pcc_eng_07_061.3782_x0975972_094:1-5-6   vocative        nothing   
pcc_eng_06_034.7520_x0545912_06:1-5-6    vocative           none   

                                          adv_form_lower adj_form_lower  \
hit_id                                                                    
nyt_eng_19981129_0157_24:48-51-52              certainly       relative   
nyt_eng_19960302_0173_15:09-22-23       constitutionally          unfit   
nyt_eng_20000806_0185_4:04-10-11                somewhat           akin   
nyt_eng_19990909_0107_2:7-8-9                        too        pleased   
pcc_eng_22_044.1204_x0696858_08:4-5-6                 so          elite   
...                                                  ...            ...   
pcc_eng_10_053.8207_x0854279_115:2-8-9              more        shocked   
pcc_eng_19_027.9735_x0435427_58:1-5-6              quite       thankful   
pcc_eng_11_057.3830_x0912194_107:2-6-7        completely           fine   
pcc_eng_07_061.3782_x0975972_094:1-5-6          probably         asleep   
pcc_eng_06_034.7520_x0545912_06:1-5-6                n't        serious   

                                                                              text_window  \
hit_id                                                                                      
nyt_eng_19981129_0157_24:48-51-52       would prove to be nothing special , certainly ...   
nyt_eng_19960302_0173_15:09-22-23       supporters admit she is nothing if not indepen...   
nyt_eng_20000806_0185_4:04-10-11        the return was nothing but a beat down _ somew...   
nyt_eng_19990909_0107_2:7-8-9           the Catholic Church being none too pleased , o...   
pcc_eng_22_044.1204_x0696858_08:4-5-6          I guess you ain't so elite after all , are   
...                                                                                   ...   
pcc_eng_10_053.8207_x0854279_115:2-8-9  And nobody , nobody could have been more shock...   
pcc_eng_19_027.9735_x0435427_58:1-5-6        Nothing , I am quite thankful for the way my   
pcc_eng_11_057.3830_x0912194_107:2-6-7            " Nothing , I 'm completely fine Emma .   
pcc_eng_07_061.3782_x0975972_094:1-5-6    Nothing , she 's probably asleep , and when she   
pcc_eng_06_034.7520_x0545912_06:1-5-6                         None , it was n't serious .   

                                       corpus  match_ix  
hit_id                                                   
nyt_eng_19981129_0157_24:48-51-52         nyt  48-51-52  
nyt_eng_19960302_0173_15:09-22-23         nyt  09-22-23  
nyt_eng_20000806_0185_4:04-10-11          nyt  04-10-11  
nyt_eng_19990909_0107_2:7-8-9             nyt     7-8-9  
pcc_eng_22_044.1204_x0696858_08:4-5-6     pcc     4-5-6  
...                                       ...       ...  
pcc_eng_10_053.8207_x0854279_115:2-8-9    pcc     2-8-9  
pcc_eng_19_027.9735_x0435427_58:1-5-6     pcc     1-5-6  
pcc_eng_11_057.3830_x0912194_107:2-6-7    pcc     2-6-7  
pcc_eng_07_061.3782_x0975972_094:1-5-6    pcc     1-5-6  
pcc_eng_06_034.7520_x0545912_06:1-5-6     pcc     1-5-6  

[10000 rows x 7 columns]

In [38]:
for deprel in deprel_counts[deprel_counts>0.001].index:
    
    # print(f'\n- **P** `{deprel}`')
    print(f'\n#### Sample of `{deprel}` relations (with `ADJ` head)\n')
    selection = df_clean[df_clean.neg_deprel==deprel]
    print(selection.sample(min(len(selection),25)).filter(regex=r'(lower|window|ix)').set_index('match_ix').to_markdown())


#### Sample of `mark` relations (with `ADJ` head)

| match_ix   | neg_form_lower   | adv_form_lower   | adj_form_lower   | text_window                                                                                                                                                                         |
|:-----------|:-----------------|:-----------------|:-----------------|:------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| 16-18-19   | without          | too              | complex          | enough to be interesting without being too complex to easily pick up                                                                                                                |
| 17-19-20   | without          | too              | progressive      | adopting a retro sound without being too progressive .                                                            


#### Sample of `mark` relations (with `ADJ` head)

| match_ix   | neg_form_lower   | adv_form_lower   | adj_form_lower   | text_window                                                                                                                                                                         |
|:-----------|:-----------------|:-----------------|:-----------------|:------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| 16-18-19   | without          | too              | complex          | enough to be interesting without being too complex to easily pick up                                                                                                                |
| 17-19-20   | without          | too              | progressive      | adopting a retro sound without being too progressive .                                                                                                                              |
| 6-8-9      | without          | overly           | delicate         | feels special and romantic without being overly delicate or stuffy .                                                                                                                |
| 30-32-33   | without          | unduly           | archaic          | give a period feel without being unduly archaic .                                                                                                                                   |
| 11-13-14   | without          | deeply           | knowledgeable    | write an Aeon essay without being deeply knowledgeable and immersed in the                                                                                                          |
| 38-40-41   | without          | inordinately     | complex          | 's very good coffee without being inordinately complex in flavor , it                                                                                                               |
| 08-10-11   | without          | too              | direct           | opportunities to engage them without being too direct .                                                                                                                             |
| 25-27-28   | without          | too              | intrusive        | of introducing Korean flavours without being too intrusive or confronting to the                                                                                                    |
| 09-12-13   | without          | incredibly       | religious        | spiritual element to it without it being incredibly religious . "                                                                                                                   |
| 1-3-4      | without          | terribly         | thick            | Without being terribly thick , it 's creamy                                                                                                                                         |
| 11-14-15   | without          | overwhelmingly   | sweet            | they 're still cakes without being too overwhelmingly sweet .                                                                                                                       |
| 18-20-21   | without          | unnecessarily    | chatty           | can hang out with without being unnecessarily chatty , the vast majority                                                                                                            |
| 29-31-32   | without          | overly           | flashy           | are memorable and stylish without being overly flashy and there is just                                                                                                             |
| 31-33-34   | without          | too              | pushy            | organization 's persona , without being too pushy .                                                                                                                                 |
| 17-19-20   | without          | too              | harsh            | right off the card without being too harsh ( black would 've                                                                                                                        |
| 19-21-22   | without          | too              | cold             | away with it , without being too cold .                                                                                                                                             |
| 18-20-21   | without          | physically       | present          | can show their solidarity without being physically present in Saint Peter 's                                                                                                        |
| 10-12-13   | without          | overly           | sweet            | tart and definitely sweet without being overly sweet .                                                                                                                              |
| 2-4-5      | without          | overwhelmingly   | dark             | Smart without being overwhelmingly dark , Crisis on Two                                                                                                                             |
| 13-40-41   | without          | too              | much             | life was so hard without her husband , how going to nursing school and working at Wal-Mart and trying to raise three children , all at the same time , was too much for her to bear |
| 13-15-16   | without          | too              | cutesy           | a touch of color without being too cutesy , a simple gold                                                                                                                           |
| 12-14-15   | without          | too              | invasive         | stuff in an apartment without being too invasive .                                                                                                                                  |
| 23-25-26   | without          | overly           | feminine         | Chic and cosmopolitan ( without being overly feminine ) - Fresh and                                                                                                                 |
| 36-38-39   | without          | too              | sweet            | lollypop and a margarita without being too sweet .                                                                                                                                  |
| 18-20-21   | without          | so               | frugal           | help me save money without being so frugal as to lead me                                                                                                                            |

#### Sample of `dep` relations (with `ADJ` head)

| match_ix   | neg_form_lower   | adv_form_lower   | adj_form_lower   | text_window                                                                                                                                                                   |
|:-----------|:-----------------|:-----------------|:-----------------|:------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| 34-35-36   | no               | less             | rational         | delay holiday purchases is no less rational than women 's tendency                                                                                                            |
| 09-10-11   | none             | as               | contentious      | was complex , but none as contentious or embittered as Ellison                                                                                                                |
| 5-6-7      | no               | more             | helpful          | voracious reading is probably no more helpful if what you covet                                                                                                               |
| 12-13-14   | not              | so               | much             | , of course , not so much an American film as                                                                                                                                 |
| 22-23-24   | not              | really           | true             | to involve villains -- not really true , but we pretend                                                                                                                       |
| 36-40-41   | not              | more             | romantic         | on seasonal cooking -LRB- not because it 's more romantic , but because it                                                                                                    |
| 28-29-30   | not              | so               | great            | Tengo -RRB- and some not so great -LRB- like Devendra Banhart                                                                                                                 |
| 08-09-10   | no               | less             | contrived        | show Thursday came off no less contrived than Hunter 's .                                                                                                                     |
| 29-30-31   | no               | less             | noble            | now conquered you for no less noble a cause . ''                                                                                                                              |
| 08-12-13   | not              | remotely         | original         | on Alright sound great not because they are remotely original but because such playing                                                                                        |
| 6-7-8      | not              | yet              | certain          | is possible , but not yet certain , these were ritual                                                                                                                         |
| 02-30-31   | not              | that             | big              | `` Not that national forest management is n't an issue , but just by the very fact of where these fires are burning -- national forest management is n't that big of a deal , |
| 10-11-12   | no               | less             | crucial          | set in Bosnia are no less crucial than they were in                                                                                                                           |
| 11-12-13   | not              | too              | hot              | think glorious weather , not too hot .                                                                                                                                        |
| 11-12-13   | no               | less             | funny            | , bu it 's no less funny .                                                                                                                                                    |
| 1-2-3      | not              | that             | big              | not that big , but a good                                                                                                                                                     |
| 04-14-15   | not              | delightfully     | wacky            | how could they not ? On first impression , Stein 's text is delightfully wacky , and Thomson 's                                                                               |
| 12-17-18   | not              | so               | popular          | '' are extraordinary -- not because the image is so popular with art thieves but                                                                                              |
| 16-17-18   | none             | too              | scrupulous       | at sea , and none too scrupulous in how he did                                                                                                                                |
| 2-3-4      | nothing          | as               | bad              | `` Nothing as bad as what they did                                                                                                                                            |
| 17-18-19   | not              | so               | endearing        | endearing -- and sometimes not so endearing -- neighbors of Haines                                                                                                            |
| 6-7-8      | not              | so               | much             | that 's a reflection not so much on the writer as                                                                                                                             |
| 10-11-12   | not              | so               | much             | of your home depends not so much on its square footage                                                                                                                        |
| 5-7-8      | nothing          | as               | exciting         | the two-pronged answer : nothing nearly as exciting , and a lot                                                                                                               |
| 01-10-11   | no               | as               | gritty           | no , these cops and ambulance jockeys are n't as gritty or dyspeptic as the                                                                                                   |

#### Sample of `conj` relations (with `ADJ` head)

| match_ix   | neg_form_lower   | adv_form_lower   | adj_form_lower   | text_window                                                                         |
|:-----------|:-----------------|:-----------------|:-----------------|:------------------------------------------------------------------------------------|
| 10-11-12   | not              | particularly     | spicy            | good -- straightforward , not particularly spicy but perfect in texture             |
| 3-7-8      | not              | as               | dark             | whether or not his heart is as dark as his comments ,                               |
| 13-14-15   | not              | too              | sweet            | to saute quickly but not too sweet for dinner .                                     |
| 7-8-9      | not              | extraordinarily  | eye-catching     | was timeless , but not extraordinarily eye-catching or cheerful .                   |
| 03-13-14   | not              | beyond           | human            | whether or not members of the Heaven 's Gate cult were `` beyond human , '' as they |
| 10-13-14   | not              | actually         | willing          | to see whether or not he 's actually willing to be responsible for                  |
| 20-23-24   | not              | legally          | responsible      | seek settlement whether or not it was legally responsible , '' he told              |
| 12-13-14   | not              | necessarily      | ideological      | red and blue , not necessarily ideological but tribal , ''                          |
| 27-29-30   | not              | as               | much             | higher only recently and not nearly as much .                                       |
| 12-14-15   | not              | so               | frightening      | Kinney explains trains _ not half so frightening as air travel and                  |
| 08-09-10   | not              | that             | much             | will react positively but not that much until we see there                          |
| 23-24-25   | not              | much             | better           | worse '' and `` not much better . ''                                                |
| 38-39-40   | not              | as               | capable          | as my mother , not as capable and independent as Maryellen                          |
| 15-16-17   | not              | very             | social           | a quiet soul and not very social .                                                  |
| 08-11-12   | not              | really           | serious          | sense of whether or not you 're really serious about this .                         |
| 16-22-23   | not              | still            | alive            | buzzed about whether or not Iraqi President Saddam Hussein was still alive .        |
| 47-48-49   | not              | highly           | literate         | non-native English speakers and not highly literate , '' Farrell said               |
| 23-24-25   | not              | readily          | treatable        | you probably realize , not readily treatable . ''                                   |
| 15-16-17   | not              | as               | gregarious       | are analytical thinkers and not as gregarious as Doran was known                    |
| 13-17-18   | not              | entirely         | convinced        | 120 days whether or not the population is entirely convinced by the degree of       |
| 23-25-26   | not              | as               | much             | too , but just not quite as much .                                                  |
| 21-22-23   | not              | yet              | dire             | with uncertain _ but not yet dire _ results .                                       |
| 26-31-32   | not              | very             | funny            | thought into whether or not the show is even very funny .                           |
| 18-19-20   | not              | particularly     | enthusiastic     | three dozen polite but not particularly enthusiastic members of the Valley          |
| 19-20-21   | not              | totally          | effective        | be `` reasonably but not totally effective , '' and it                              |

#### Sample of `preconj` relations (with `ADJ` head)

| match_ix   | neg_form_lower   | adv_form_lower   | adj_form_lower   | text_window                                                                          |
|:-----------|:-----------------|:-----------------|:-----------------|:-------------------------------------------------------------------------------------|
| 12-13-14   | neither          | too              | firm             | the rice cake is neither too firm nor too loose ,                                    |
| 17-19-20   | neither          | most             | favorable        | adding that Arizona is neither the most favorable nor the least attractive           |
| 16-17-18   | neither          | as               | infamous         | of course _ is neither as infamous nor as polarizing as                              |
| 19-20-21   | neither          | readily          | understandable   | of law that is neither readily understandable nor easy to apply                      |
| 15-16-17   | neither          | as               | bleak            | the new disc is neither as bleak as 1997 's Time                                     |
| 15-16-17   | neither          | as               | shocking         | a little predictable and neither as shocking nor as funny as                         |
| 7-8-9      | neither          | too              | optimistic       | of the economy are neither too optimistic or pessimistic at the                      |
| 7-8-9      | neither          | as               | airy             | The American '' is neither as airy as some French films                              |
| 5-6-7      | neither          | more             | correct          | but the message is neither more correct nor more acceptable for                      |
| 10-11-12   | neither          | as               | immediate        | by Asian nations are neither as immediate nor as far-reaching as                     |
| 5-6-7      | neither          | too              | hot              | `` The economy is neither too hot nor too cold and                                   |
| 49-50-51   | neither          | as               | safe             | that the device was neither as safe nor as effective as                              |
| 19-20-21   | neither          | too              | hot              | '' economy that is neither too hot nor too cold .                                    |
| 13-14-15   | neither          | precisely        | french           | forest , it feels neither precisely French nor American .                            |
| 16-17-18   | neither          | fully            | female           | as someone who seemed neither fully female nor male , even                           |
| 1-2-3      | neither          | as               | flowery          | neither as flowery as an Alsatian riesling                                           |
| 6-7-8      | neither          | as               | mild             | governors ' program is neither as mild as programs proposed by                       |
| 4-5-6      | neither          | too              | tired            | but McGrath is neither too tired nor too shocked to                                  |
| 5-6-7      | neither          | as               | desirable        | these late works are neither as desirable nor as rare .                              |
| 09-10-11   | neither          | as               | special          | and our media are neither as special , nor as cursed                                 |
| 24-25-26   | neither          | too              | hot              | with Goldilocks climates -- neither too hot nor too cold .                           |
| 27-29-30   | neither          | sexually         | immoral          | International version : `` Neither the sexually immoral nor idolaters nor adulterers |
| 7-8-9      | neither          | heavily          | intoxicated      | said the woman was neither heavily intoxicated nor unconscious .                     |
| 4-5-6      | neither          | as               | sharp            | but it was neither as sharp as the prosecution wanted                                |
| 47-48-49   | neither          | as               | distinct         | American art will seem neither as distinct nor as dominant as                        |

#### Sample of `parataxis` relations (with `ADJ` head)

| match_ix   | neg_form_lower   | adv_form_lower   | adj_form_lower   | text_window                                                                                                                                                                                  |
|:-----------|:-----------------|:-----------------|:-----------------|:---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| 04-12-13   | not              | less             | likely           | typically , though not always , the black job seekers were less likely to be invited for                                                                                                     |
| 35-39-40   | not              | more             | interesting      | Sharks became _ if not better _ much more interesting .                                                                                                                                      |
| 37-41-42   | not              | certainly        | inappropriate    | Act are , if not illegal , are certainly inappropriate . ''                                                                                                                                  |
| 37-41-42   | not              | less             | cautious         | to be , if not overexuberant , maybe less cautious . ''                                                                                                                                      |
| 06-11-12   | not              | increasingly     | visible          | phone use , though not widespread , was once increasingly visible among North Koreans .                                                                                                      |
| 04-12-13   | not              | necessarily      | noticeable       | and , while not always invisible , nerve agents are not necessarily noticeable .                                                                                                             |
| 20-31-32   | not              | nonetheless      | real             | -RRB- that , while not as apparent as they are on most teams , are nonetheless real .                                                                                                        |
| 11-13-14   | few              | truly            | genuine          | if it 's very few are truly genuine and hope to meet                                                                                                                                         |
| 44-50-51   | few              | pretty           | sure             | which are just a few ) , you can be pretty sure that the rest of                                                                                                                             |
| 07-22-23   | not              | nonetheless      | unusual          | career path , though not as spectacular as the president 's rise to power from rural poverty , is nonetheless unusual .                                                                      |
| 08-16-17   | not              | relatively       | unusual          | as this , though not uncommon in the United States , are relatively unusual in France .                                                                                                      |
| 07-11-12   | not              | notoriously      | impervious       | for months , if not years , are notoriously impervious to outside events .                                                                                                                   |
| 19-27-28   | not              | much             | wider            | , which , while not on as desirable a block , is much wider and allows for much                                                                                                              |
| 10-16-17   | not              | less             | obscene          | will become , if not less exorbitant , at least less obscene .                                                                                                                               |
| 08-12-13   | not              | hardly           | unprecedented    | by rowboat , though not routine , is hardly unprecedented .                                                                                                                                  |
| 05-25-26   | not              | so               | high             | after all , if not for a couple of quirky scenarios , the ring count for the Spurs ' forward probably would n't be so high .                                                                 |
| 07-13-14   | not              | less             | dramatic         | real-life story , while not so tidy , is no less dramatic .                                                                                                                                  |
| 05-09-10   | not              | surprisingly     | accurate         | prediction markets , while not perfect , are surprisingly accurate -- certainly more accurate                                                                                                |
| 07-11-12   | not              | hardly           | aware            | local governments , if not all , are hardly aware of these laws .                                                                                                                            |
| 13-18-19   | not              | less             | gloomy           | results that , while not strong overall , were less gloomy than expected .                                                                                                                   |
| 06-26-27   | not              | still            | insidious        | These pressures , while not as stark as the murder , detention and torture of journalists in places like Algeria and Peru , are still insidious and dangerous . ''                           |
| 08-14-15   | not              | as               | bad              | bad taste , if not sacrilege , this one 's as bad as it gets .                                                                                                                               |
| 06-10-11   | not              | truly            | rare             | an admission , while not unprecedented , is truly rare among presidential candidates .                                                                                                       |
| 37-49-50   | not              | more             | willing          | acquaintances who , though not necessarily beholden to him in a legal sense , were certainly more willing to accede to his                                                                   |
| 08-30-31   | not              | that             | remarkable       | historical context , if not in the hothouse of congressional politics , the Republicans ' departure from the New Deal and Great Society is n't all that remarkable ; Republicans fought both |

#### Sample of `aux` relations (with `ADJ` head)

| match_ix   | neg_form_lower   | adv_form_lower   | adj_form_lower   | text_window                                                           |
|:-----------|:-----------------|:-----------------|:-----------------|:----------------------------------------------------------------------|
| 2-4-5      | ain't            | so               | concerned        | You ain't been so concerned with how dirty mine                       |
| 11-12-13   | ain't            | even             | worried          | in them that they ain't even worried about the scoreboard .           |
| 2-3-4      | ain't            | really           | happy            | We ain't really happy here .                                          |
| 29-30-31   | ain't            | never            | true             | I bet that they ain't never true ...                                  |
| 21-22-23   | ain't            | already          | deep             | be in that I ain't already deep in Take it how                        |
| 41-42-43   | aint             | even             | mad              | need some encouragement i aint even mad at you do whats               |
| 1-2-3      | aint             | even             | worth            | Aint even worth the bother to list                                    |
| 12-13-14   | ain't            | hardly           | real             | Q-tip cotton Ya'll dudes ain't hardly real Ya boys spit on            |
| 61-62-63   | ain't            | even             | close            | burrito bowl , it ain't even close .                                  |
| 10-11-12   | ain't            | even             | ashamed          | running around like they ain't even ashamed .                         |
| 6-8-9      | ain't            | so               | dumb             | other words , they ain't necessarily so dumb .                        |
| 18-19-20   | ain't            | so               | rare             | turn over ; wot ain't so rare .                                       |
| 14-19-20   | ain't            | long             | enough           | ring , but you ain't been in the league long enough to talk to people |
| 08-09-10   | ain't            | too              | worried          | personal opinion which I ain't too worried about .                    |
| 2-3-4      | ain't            | just             | soft             | You ain't just soft in the head ,                                     |
| 3-4-5      | ain't            | really           | good             | But you ain't really good                                             |
| 11-12-13   | ain't            | even             | black            | 's addiction those who ain't even black use it .                      |
| 16-17-18   | ain't            | quite            | ready            | but I calculate we ain't quite ready yet .                            |
| 33-34-35   | ain't            | even             | close            | fact is that you ain't even close .                                   |
| 10-12-13   | ain't            | that             | active           | So my pal Ed ain't been that active in the movies since               |
| 7-8-9      | ain't            | even             | close            | day in the sun ain't even close                                       |
| 36-37-38   | ain't            | really           | scared           | comfortable , now you ain't really scared                             |
| 08-09-10   | ain't            | just             | disposable       | you think , people ain't just disposable .                            |
| 09-11-12   | ain't            | so               | bad              | , working with you ain't been so bad .                                |
| 28-29-30   | ain't            | too              | old              | ain't dead , you ain't too old ! ) ?                                  |

#### Sample of `dislocated` relations (with `ADJ` head)

| match_ix   | neg_form_lower   | adv_form_lower   | adj_form_lower   | text_window                                                                                                        |
|:-----------|:-----------------|:-----------------|:-----------------|:-------------------------------------------------------------------------------------------------------------------|
| 2-8-9      | nothing          | so               | sorry            | Was nothing against you , I 'm so sorry if you took it                                                             |
| 04-11-12   | nothing          | sublimely        | happy            | " Oh , nothing dear , I 'm just so sublimely happy , that 's all                                                   |
| 2-6-7      | nothing          | just             | boring           | " Nothing , they 're just boring .                                                                                 |
| 01-12-13   | nothing          | n't              | worth            | Nothing short of complete dedication , otherwise , the costumes are n't worth the work ( in                        |
| 03-09-10   | none             | all              | original         | No , none of that , it 's all original .                                                                           |
| 01-18-19   | nothing          | much             | better           | Nothing cooked in a wood fired oven tastes like food from a conventional oven , it 's much better .                |
| 1-8-9      | nothing          | not              | interested       | Nothing but a career , it is not interested .                                                                      |
| 05-11-12   | nothing          | mostly           | free             | Besides the rig , nothing , the miner programs are mostly free .                                                   |
| 2-7-8      | nothing          | so               | happy            | " Nothing , I 'm just so happy . "                                                                                 |
| 01-16-17   | nothing          | chronically      | undecided        | Nothing too important , just how I wanted to work in aviation , Liam was chronically undecided and Cooper was torn |
| 01-12-13   | none             | too              | big              | None , we take showers only , the whirlpool tub is too big and our water heater                                    |
| 3-8-9      | none             | n't              | capable          | Why , none ; the screenwriter is n't capable of coming up with                                                     |
| 08-12-13   | nothing          | not              | qualified        | doc replies , " nothing , you 're not qualified . "                                                                |
| 1-8-9      | nothing          | very             | nice             | Nothing against them , they 're all very nice , but I just                                                         |
| 01-14-15   | nothing          | very             | smart            | Nothing against the team that were working on it , they were all very smart , but like I                           |
| 02-10-11   | nothing          | very             | busy             | " nothing my dear , it just I 'm very busy lately "                                                                |

#### Sample of `rel` relations (with `ADJ` head)

| match_ix   | neg_form_lower   | adv_form_lower   | adj_form_lower   | text_window                                                                                                                                   |
|:-----------|:-----------------|:-----------------|:-----------------|:----------------------------------------------------------------------------------------------------------------------------------------------|
| 13-19-20   | without          | incomparably     | superb           | the free market , without which America would not be incomparably superb , except for its                                                     |
| 09-17-18   | without          | as               | blue             | he projected a vigor without which the blues could not be half as blue .                                                                      |
| 37-41-42   | without          | very             | difficult        | India in particular , without whom it is very difficult to see how we                                                                         |
| 27-33-34   | without          | little           | more             | with that toasty flavor without which a sparkling wine is little more than effervescent grape juice                                           |
| 33-49-50   | without          | vastly           | poorer           | expect from Heyday , without whose journal `` News From Native California '' our knowledge of pre-contact California would be vastly poorer . |
| 11-17-18   | without          | as               | successful       | have a distinctive culture without which they would n't be as successful , '' he said                                                         |
| 33-40-41   | without          | as               | dull             | and general sleaziness , without which the primary season would be as dull as `` Montel Williams                                              |
| 09-16-17   | without          | even             | higher           | out significant structural adjustments without which the deficit would 've been even higher , '' said Almunia                                 |
| 27-40-41   | without          | as               | rich             | , hokey dignity - without which the lifestyles of the rich and famous just would n't be as rich or as famous .                                |
| 22-27-28   | without          | extremely        | vulnerable       | Khmer Rouge guerrillas , without whom they would be extremely vulnerable militarily .                                                         |
| 14-19-20   | without          | more             | difficult        | of the world , without which it becomes ever more difficult to execute foreign policy                                                         |
| 17-23-24   | without          | almost           | impossible       | of clean-energy technologies , without which major reductions would be almost impossible to achieve .                                         |
| 12-17-18   | without          | much             | shorter          | ? d montages , without which it would be much shorter .                                                                                       |


In [25]:
print(df_clean[df_clean.neg_deprel=='preconj'].sample(20).filter(regex=r'(lower|window)').reset_index(drop=True).to_markdown())

|    | neg_form_lower   | adv_form_lower   | adj_form_lower   | text_window                                                                              |
|---:|:-----------------|:-----------------|:-----------------|:-----------------------------------------------------------------------------------------|
|  0 | neither          | especially       | patient          | winning : She 's neither English nor especially patient , but this industrious           |
|  1 | neither          | even             | sneaky           | , his application was neither illegal nor even sneaky .                                  |
|  2 | neither          | too              | dry              | dark closet that is neither too dry nor humid .                                          |
|  3 | neither          | especially       | old              | switch , which was neither especially old nor subject to enormous                        |
|  4 | neither          | especially       | funny            | polite

|    | neg_form_lower   | adv_form_lower   | adj_form_lower   | text_window                                                                       |
|---:|:-----------------|:-----------------|:-----------------|:----------------------------------------------------------------------------------|
|  0 | neither          | too              | hot              | an economy that proved neither too hot nor too cold .                             |
|  1 | neither          | even             | informal         | as normal English , neither slang nor even informal .                             |
|  2 | neither          | as               | cheap            | have suffered by being neither as cheap as the discounters nor                    |
|  3 | neither          | as               | targeted         | tax cut that is neither as targeted as the Democrats would                        |
|  4 | neither          | suitably         | christian        | implication that he is neither suitably Christian nor American in his             |
|  5 | neither          | excruciatingly   | thick            | Eddie Kean , was neither excruciatingly thick like Mortimer nor wickedly          |
|  6 | neither          | too              | tired            | but McGrath is neither too tired nor too shocked to                               |
|  7 | neither          | particularly     | tender           | braised lamb shank is neither particularly tender nor particularly flavorful ,    |
|  8 | neither          | too              | fast             | a rate that 's neither too fast nor too slow .                                    |
|  9 | neither          | overtly          | supportive       | released a prepared statement neither overtly supportive or critical .            |
| 10 | neither          | as               | severe           | the coming weeks is neither as severe as those in many                            |
| 11 | neither          | too              | hard             | Goldilocks charm of being neither too hard nor too soft ,                         |
| 12 | neither          | too              | academic         | the insightful analyses are neither too academic nor pedantic and are             |
| 13 | neither          | officially       | catholic         | the 300-member parish was neither officially Catholic nor any longer Episcopalian |
| 14 | neither          | as               | common           | general practitioners are neither as common nor as general as                     |
| 15 | neither          | too              | hot              | where conditions should be neither too hot nor too cold but                       |
| 16 | neither          | too              | sumptuous        | lived-in rooms that looked neither too sumptuous nor too new ,                    |
| 17 | neither          | as               | good             | available at supermarkets was neither as good as the fries sold                   |
| 18 | neither          | too              | regular          | prefer patterns that are neither too regular , like the test                      |
| 19 | neither          | as               | harmful          | a drug that is neither as harmful nor as addictive as                             |
